op-acceptance-tests/tests/isthmus/pectra_features_test.go

```diff
@@ -0,0 +1,262 @@
+package isthmus
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/testlib/validators"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/types"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-service/txplan"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm/program"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// Test vectors for Pectra features
+	SET_CODE_TX_BASIC uint64 = iota
+	BLOCK_HISTORY_CONSISTENCY
+	EMPTY_REQUESTS_HASH
+	CALLDATA_COST_INCREASE
+)
+
+// TestPectra verifies that L1 Pectra inherited features function properly on the OP Stack.
+func TestPectra(t *testing.T) {
+	// Define which L2 chain we'll test.
+	chainIdx := uint64(0)
+
+	// Get validators and getters for accessing the system and wallets
+	walletGetter, walletValidator := validators.AcquireL2WalletWithFunds(chainIdx, types.NewBalance(big.NewInt(params.Ether)))
+
+	// Run isthmus tests
+	_, forkValidator := validators.AcquireL2WithFork(chainIdx, rollup.Isthmus)
+
+	t.Run("SetCodeBasic", func(t *testing.T) {
+		systest.SystemTest(t,
+			pectraFeatureTest(walletGetter, chainIdx, SET_CODE_TX_BASIC),
+			walletValidator,
+			forkValidator,
+		)
+	})
+	t.Run("BlockHistoryConsistency", func(t *testing.T) {
+		systest.SystemTest(t,
+			pectraFeatureTest(walletGetter, chainIdx, BLOCK_HISTORY_CONSISTENCY),
+			walletValidator,
+			forkValidator,
+		)
+	})
+	t.Run("EmptyRequestsHash", func(t *testing.T) {
+		systest.SystemTest(t,
+			pectraFeatureTest(walletGetter, chainIdx, EMPTY_REQUESTS_HASH),
+			walletValidator,
+			forkValidator,
+		)
+	})
+	t.Run("CalldataCostIncrease", func(t *testing.T) {
+		systest.SystemTest(t,
+			pectraFeatureTest(walletGetter, chainIdx, CALLDATA_COST_INCREASE),
+			walletValidator,
+			forkValidator,
+		)
+	})
```

Comment:
Just an idea, in case you hadn't already considered it:

```suggestion

    testCases := []struct{
	name string
	vector uint64
    }{
	{
		name: "SetCodeBasic",
		vector: SET_CODE_TX_BASIC,
	},
	{
		name: "BlockHistoryConsistency",
		vector: BLOCK_HISTORY_CONSISTENCY,
	},
	{
		name: "EmptyRequestsHash",
		vector: EMPTY_REQUESTS_HASH,
	},
	{
		name: "CalldataCostIncrease",
		vector: CALLDATA_COST_INCREASE,
	},
    }
    
    for _,test := range testCases {
	    t.Run(test.name, func(t *testing.T) {
		    systest.SystemTest(t,
			pectraFeatureTest(walletGetter, chainIdx, test.vector),
			walletValidator,
			forkValidator,
		)
	})
```